homeassistant/components/broadlink/device.py

```diff
@@ -63,6 +63,12 @@ async def async_update(hass, entry):
         device_registry.async_update_device(device_entry.id, name=entry.title)
         await hass.config_entries.async_reload(entry.entry_id)
 
+    def _auth_fetch_firmware(self):
+        """Auth and fetch firmware."""
+        self.api.auth()
+        with suppress(BroadlinkException, OSError):
+            return self.api.get_fwversion()
```

Comment:
We should return a value explicitly outside the context manager too.
```suggestion
            return self.api.get_fwversion()
        return None
```